bpf/lib/trace.h

```diff
@@ -151,29 +155,28 @@ send_trace_notify(struct __ctx_buff *ctx, __u8 obs_point, __u32 src, __u32 dst,
 	if (!monitor)
 		monitor = TRACE_PAYLOAD_LEN;
 
-	__u64 ctx_len = (__u64)ctx_full_len(ctx);
-	__u64 cap_len = min((__u64)monitor, (__u64)ctx_len);
-	__u32 hash = get_hash_recalc(ctx);
-	struct trace_notify msg = {
-		.type = CILIUM_NOTIFY_TRACE,
-		.subtype = obs_point,
-		.source = EVENT_SOURCE,
-		.hash = hash,
-		.len_orig = ctx_len,
-		.len_cap = cap_len,
-		.version = NOTIFY_CAPTURE_VER,
-		.src_label = src,
-		.dst_label = dst,
-		.dst_id = dst_id,
-		.reason = reason,
-		.ipv6 = 0,
-		.pad = 0,
-		.ifindex = ifindex,
-		.orig_ip4 = 0,
-		.orig_pad1 = 0,
-		.orig_pad2 = 0,
-		.orig_pad3 = 0,
-	};
+	msg.type = CILIUM_NOTIFY_TRACE;
+	msg.subtype = obs_point;
+	msg.source = EVENT_SOURCE;
+	msg.version = NOTIFY_CAPTURE_VER;
```

Comment:
can't help but wonder if there's some better code refactoring/sharing we could do when reviewing ~three~ four separate, very similar, initializers like this ;)

Reply:
I'll make it a helper to simplify all the other types like this and address the other comment from above.